src/Orchard.Web/Modules/Orchard.OpenId/Controllers/AccessController.cs

```diff
@@ -32,18 +31,26 @@ public class AccessController : Controller
             _userManager = userManager;
         }
 
-        [HttpPost]
+        [HttpPost, Produces("application/json")]
         public Task<IActionResult> Token()
         {
             var request = HttpContext.GetOpenIdConnectRequest();
 
             if (request.IsPasswordGrantType())
+            {
                 return ExchangePasswordGrantType(request);
+            }
             else if (request.IsClientCredentialsGrantType())
+            {
                 return ExchangeClientCredentialsGrantType(request);
+            }
             else
             {
-                return Task.FromResult(BadRequest() as IActionResult);
+                return Task.FromResult<IActionResult>(BadRequest(new OpenIdConnectResponse
```

Comment:
Unrelated but recommended change (with the current code, you'd get an empty 400 response, which would be invalid)